spring-cloud-deployer-resource-docker/src/main/java/org/springframework/cloud/deployer/resource/docker/DockerResourceLoader.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.resource.docker;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A {@link ResourceLoader} that loads {@link DockerResource}s from locations of the format
+ * {@literal docker:<repository>:<tag>} where the value for "repository:tag" conforms to the rules
+ * for referencing Docker images.
+ *
+ * @author Thomas Risberg
+ */
+public class DockerResourceLoader  implements ResourceLoader {
```

Comment:
would this eventually include information about one or more particular registries?... and auth credentials if necessary 

assuming we eventually implement the resource's `getInputStream()`, we would need to pass those properties along to the `DockerResource` that is loaded - similar to how `MavenProperties` are passed to a `MavenResource` from the `MavenResourceLoader`... and whereas the Maven implementation uses the Aether library, we would use the Docker Java API